File: src/main/java/com/giffardtechnologies/restdocs/domain/RequestBody.java
package com.giffardtechnologies.restdocs.domain;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class RequestBody {
	private String description;
	@SerializedName("content types")
	private ArrayList<String> contentTypes;
	
	public String getDescription() {
		return description;
	}
	
	public void setDescription(String description) {
		this.description = description;
	}

	public ArrayList<String> getContentTypes() {
		return contentTypes;
	}

	public void setContentTypes(ArrayList<String> contentTypes) {
		this.contentTypes = contentTypes;
	}
}

File: src/main/java/com/giffardtechnologies/restdocs/storage/RequestBody.kt
package com.giffardtechnologies.restdocs.storage

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Describes the body that a client sends with an HTTP request.
 *
 * @property description Human-readable description of the expected request body.
 * @property contentTypes The MIME types accepted for the request body (e.g., `"application/json"`).
 */
class RequestBody {
    val description: String? = null

    @field:JsonProperty("content types")
    val contentTypes: ArrayList<String>? = null
}